milli/src/update/index_documents/mod.rs

```diff
@@ -172,6 +176,139 @@ where
         Ok((self, Ok(indexed_documents)))
     }
 
+    #[tracing::instrument(level = "trace", skip_all, target = "indexing::documents")]
+    pub fn edit_documents(
+        self,
+        documents: &RoaringBitmap,
+        context: Option<Object>,
+        code: &str,
+    ) -> Result<(Self, StdResult<(u64, u64), UserError>)> {
+        // Early return when there is no document to edit
+        if documents.is_empty() {
+            return Ok((self, Ok((0, 0))));
+        }
+
+        fn rhaimap_to_object(map: rhai::Map) -> Object {
+            let mut output = Object::new();
+            for (key, value) in map {
+                let value = serde_json::to_value(&value).unwrap();
+                output.insert(key.into(), value);
+            }
+            output
+        }
+
+        // Setup the security and limits of the Engine
+        let mut engine = Engine::new();
+        engine.set_optimization_level(OptimizationLevel::Full);
+        engine.set_max_call_levels(1000);
+        // It is an arbitrary value. We need to let users define this in the settings.
+        engine.set_max_operations(1_000_000);
+        engine.set_max_variables(1000);
+        engine.set_max_functions(30);
+        engine.set_max_expr_depths(100, 1000);
+        engine.set_max_string_size(1024 * 1024 * 1024); // 1 GiB
+        engine.set_max_array_size(10_000);
+        engine.set_max_map_size(10_000);
+
+        let ast = engine.compile(code).map_err(UserError::DocumentEditionCompilationError)?;
+        let fields_ids_map = self.index.fields_ids_map(self.wtxn)?;
+        let primary_key = self.index.primary_key(self.wtxn)?.unwrap();
+        let mut documents_batch_builder = tempfile::tempfile().map(DocumentsBatchBuilder::new)?;
+        let mut documents_to_remove = RoaringBitmap::new();
+
+        let context: Dynamic = match context {
+            Some(context) => {
+                serde_json::from_value(context.into()).map_err(InternalError::SerdeJson)?
+            }
+            None => Dynamic::from(()),
+        };
```

Comment:
Do not declare the context if it doesn't exists and do not set the context. Set it to None.